### DISCUSSION TITLE
Add automatic terraforming for building construction

### DISCUSSION THREAD

**Discussion by MaximDementyev** [voxeltycoon/issues#480] [General]
**Is your feature request related to a problem? Please describe.**
if you try to build a building on an uneven surface, the game does not do it

**Describe the solution you'd like**
automatically terraform the ground around under the building.

**Describe alternatives you've considered**
add buttons increase/decrease the height of the building base

**Additional context**
![2019-06-28_22-24-23](https://user-images.githubusercontent.com/15971031/60366282-c1db7b00-99f3-11e9-9dc0-4d5a6358bc3a.png)

**Comment by Redominus** [comment]
I would suggest having to maintain a key pressed to activate the auto flatten feature. It would try to level everything at the same elevation where the cursor is.
It could be used for rails and roads also. On rails and roads it would try to flatten to same heigth of the starting point.
